server/protocol.c

```diff
@@ -1583,6 +1596,14 @@ request_rec *ap_read_request(conn_rec *conn)
      */
     ap_add_input_filter_handle(ap_http_input_filter_handle,
                                NULL, r, r->connection);
+    if (r->proto_num <= HTTP_VERSION(1,1)) {
+        ap_add_input_filter_handle(ap_http1_body_in_filter_handle,
+                                   NULL, r, r->connection);
+        if (r->proto_num >= HTTP_VERSION(1,0)
+            && apr_table_get(r->headers_in, "Transfer-Encoding")) {
+            r->body_indeterminate = 1;
+        }
+    }
```

Comment:
Using the new filter and flag on `ap_read_request()` for the proper HTTP version.